pkg/blueprint/customizations.go

```diff
@@ -439,3 +440,10 @@ func (c *Customizations) GetRHSM() *RHSMCustomization {
 	}
 	return c.RHSM
 }
+
+func (c *Customizations) GetCACerts() *CACustomization {
+	if c == nil {
+		return nil
```

Comment:
I wonder if it wouldn't be better to also validate here, this would avoid the panic for invalid certs later. Here we could follow the pattern from e.g. GetRepositories() and return (*CACustomization,error) and then the callers (in fedora for example) can report a meaningful error for invalid cert strings early. Wdyt?

Reply:
I can do that, just note this introduces a dependency on distro/manifest package.

Reply:
Ah it is an import cycle actually, I cannot do that, do you want me to do it? I would suggest to create an explicit `CheckCACerts` function and extract the parsing X509 code into a separate package.

Reply:
Thanks for looking into this, I'm not sure I follow, maybe I'm missing something but I was thinking of something like this:
```go
diff --git a/pkg/blueprint/customizations.go b/pkg/blueprint/customizations.go
index 0bb5bacbe..f0d800077 100644
--- a/pkg/blueprint/customizations.go
+++ b/pkg/blueprint/customizations.go
@@ -428,12 +428,12 @@ func (c *Customizations) GetRHSM() *RHSMCustomization {
        return c.RHSM
 }
 
-func (c *Customizations) CheckCACerts() error {
+func (c *Customizations) checkCACerts() error {
        if c == nil {
                return nil
        }
 
-       if c.CACerts == nil {
+       if c.CACerts != nil {
                for _, bundle := range c.CACerts.PEMCerts {
                        _, err := cert.ParseCerts(bundle)
                        if err != nil {
@@ -445,9 +445,14 @@ func (c *Customizations) CheckCACerts() error {
        return nil
 }
 
-func (c *Customizations) GetCACerts() *CACustomization {
+func (c *Customizations) GetCACerts() (*CACustomization, error) {
        if c == nil {
-               return nil
+               return nil, nil
        }
-       return c.CACerts
+
+       if err := c.checkCACerts(); err != nil {
+               return nil, err
+       }
+
+       return c.CACerts, nil
 }
diff --git a/pkg/distro/fedora/images.go b/pkg/distro/fedora/images.go
index b0e737468..4c04a06c4 100644
--- a/pkg/distro/fedora/images.go
+++ b/pkg/distro/fedora/images.go
@@ -226,10 +226,11 @@ func osCustomizations(
        osc.Files = append(osc.Files, imageConfig.Files...)
        osc.Directories = append(osc.Directories, imageConfig.Directories...)
 
-       if err := c.CheckCACerts(); err != nil {
+       ca, err := c.GetCACerts()
+       if err != nil {
                panic(fmt.Sprintf("unexpected error checking CA certs: %v", err))
        }
-       if ca := c.GetCACerts(); ca != nil {
+       if ca != nil {
                osc.CACerts = ca.PEMCerts
        }
 
diff --git a/pkg/distro/rhel/images.go b/pkg/distro/rhel/images.go
index 5e13afe7f..806463b92 100644
--- a/pkg/distro/rhel/images.go
+++ b/pkg/distro/rhel/images.go
@@ -271,10 +271,11 @@ func osCustomizations(
                osc.NoBLS = *imageConfig.NoBLS
        }
 
-       if err := c.CheckCACerts(); err != nil {
+       ca, err := c.GetCACerts()
+       if err != nil {
                panic(fmt.Sprintf("unexpected error checking CA certs: %v", err))
        }
-       if ca := c.GetCACerts(); ca != nil {
+       if ca != nil {
                osc.CACerts = ca.PEMCerts
        }
 
```
i.e. very time we call GetCACerts() we implicitly check as well - this way we will not forget and it's similar to the pattern of GetRepositories() ?

Reply:
This is added now, marking it as resolved (and THANK YOU)